app/src/main/java/com/example/travellens/fragments/PostsFragment.java

```diff
@@ -265,17 +268,18 @@ private void setUpCustomChip(String text) {
         cgFilter.setLayoutParams(new ViewGroup.LayoutParams(cgFilter.getLayoutParams().width, ViewGroup.LayoutParams.WRAP_CONTENT));
         Chip chip = new Chip(getContext());
         chip.setText(text);
-        /* TODO change UI on chips dynamically */
+
         // adding to my list so i can use it to filter later
-        wordsToFilterBy.add(text);
+        wordsToFilterBy.add(text.toUpperCase(Locale.ROOT).replace(' ', '_'));
 
         chip.setCloseIconVisible(true);
         cgFilter.addView(chip);
         chip.setOnCloseIconClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 cgFilter.removeView(v);
-                wordsToFilterBy.remove(chip.getText());
+                wordsToFilterBy.remove(chip.getText().toString().toUpperCase(Locale.ROOT)
+                        .replace(' ','_'));
```

Comment:
What's the reason for this?

Also, it's being done at least 4 different times that I see. Should create a common method for it.

Reply:
The reason I did this is because the filter types given by the API are listed like AIRPORT and TRAVEL_AGENCY; I didn't want it to display like this which is why I have to transform the displayed string by replacing the spaces with underscores.